src/core/interaction-creator.ts

```diff
@@ -0,0 +1,194 @@
+                    ? {...queryOfRequest?.matchingRules, ...requestBody?.matchingRules}
+                    : undefined;
+
+            return newInteraction;
+        }
+        throw Error;
+    }
+
+    private getResponseBodyForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const functionBody = apiFunctionNode.getFirstChildByKindOrThrow(ts.SyntaxKind.Block);
+        let responseBodyType = getResponseTypeFromFunctionBody(functionBody)?.getType();
+        if (responseBodyType === undefined) {
+            responseBodyType = getReturnTypeOfFunction(apiFunctionNode.getType());
+        }
+        const basicTypeRepresentationOfResponse = getBasicRepresentationOfType(responseBodyType, this.sourceFile, true);
+        return {
+            body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfResponse),
+            matchingRules: mapMatchingRepresentation(
+                changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfResponse, '$.body'),
+                '$.body',
+            ),
+        };
+    }
+
+    private getRequestBodyForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const requestBodyElement =
+            getParameterWithJsDocFromFunction(apiFunctionNode, 'pact-request-body') ||
+            getVariableWithJsDocFromFunction(apiFunctionNode, 'pact-request-body');
+        if (requestBodyElement) {
+            const requestBodyElementType = requestBodyElement.getType();
+            const basicTypeRepresentationOfRequestBody = getBasicRepresentationOfType(requestBodyElementType, this.sourceFile);
+            return {
+                body: changeObjectRepresentationIntoExample(basicTypeRepresentationOfRequestBody),
+                matchingRules: mapMatchingRepresentation(
+                    changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfRequestBody, '$.body'),
+                    '$.body',
+                ),
+            };
+        }
+    }
+
+    private getQueryRequestForApiFunction(apiFunctionNode: tsMorph.Node) {
+        const queryElement =
+            getParameterWithJsDocFromFunction(apiFunctionNode, 'pact-query') ||
+            getVariableWithJsDocFromFunction(apiFunctionNode, 'pact-query');
+        if (queryElement) {
+            const queryElementType = queryElement.getType();
+            const basicTypeRepresentationOfRequestBody = getBasicRepresentationOfType(queryElementType, this.sourceFile);
+            const exampleRepresentationOfQueryObject = changeObjectRepresentationIntoExample(basicTypeRepresentationOfRequestBody);
+            return {
+                query: qs.stringify(exampleRepresentationOfQueryObject),
+                matchingRules: mapMatchingRepresentation(
+                    changeObjectRepresentationIntoMatchingRules(basicTypeRepresentationOfRequestBody, '$.query'),
+                    '$.query',
+                ),
+            };
+        }
+    }
+}
+
+const getFunctionNode = (node: tsMorph.Node) => {
```

Comment:
either make those static methods of `InteractionCreator`, either move to separate module

Reply:
I've moved them into class as static methods.